tutorials/general/t8_tutorial_build_cmesh.cxx

```diff
@@ -0,0 +1,538 @@
+    1, 0.5, 0,
+    0, 0.5, 0,                  /* tree 3, quad */
+    0.5, 0.5, 0,
+    0, 1, 0,
+    0.5, 1, 0,
+    0.5, 0.5, 0,                /* tree 4, triangle */
+    1, 0.5, 0,
+    1, 1, 0,
+    0.5, 0.5, 0,                /* tree 5, triangle */
+    1, 1, 0,
+    0.5, 1, 0
+  };
+
+  /* 2. Initialization of the mesh */
+  t8_cmesh_t          cmesh;
+  t8_cmesh_init (&cmesh);
+
+  /* 3. Definition of the geometry */
+  t8_geometry_c      *linear_geom = t8_geometry_linear_new (2);
+  t8_cmesh_register_geometry (cmesh, linear_geom);      /* Use linear geometry */
+
+  /* 4. Definition of the classes of the different trees */
+  t8_cmesh_set_tree_class (cmesh, 0, T8_ECLASS_TRIANGLE);
+  t8_cmesh_set_tree_class (cmesh, 1, T8_ECLASS_TRIANGLE);
+  t8_cmesh_set_tree_class (cmesh, 2, T8_ECLASS_QUAD);
+  t8_cmesh_set_tree_class (cmesh, 3, T8_ECLASS_QUAD);
+  t8_cmesh_set_tree_class (cmesh, 4, T8_ECLASS_TRIANGLE);
+  t8_cmesh_set_tree_class (cmesh, 5, T8_ECLASS_TRIANGLE);
+
+  /* 5. Classification of the vertices for each tree */
+  t8_cmesh_set_tree_vertices (cmesh, 0, vertices, 3);
+  t8_cmesh_set_tree_vertices (cmesh, 1, vertices + 9, 3);
+  t8_cmesh_set_tree_vertices (cmesh, 2, vertices + 18, 4);
+  t8_cmesh_set_tree_vertices (cmesh, 3, vertices + 30, 4);
+  t8_cmesh_set_tree_vertices (cmesh, 4, vertices + 42, 3);
+  t8_cmesh_set_tree_vertices (cmesh, 5, vertices + 51, 3);
+
+  /* 6. Definition of the face neighbors between the different trees */
+  t8_cmesh_set_join (cmesh, 0, 1, 1, 2, 0);
+  t8_cmesh_set_join (cmesh, 0, 2, 0, 0, 0);
+  t8_cmesh_set_join (cmesh, 0, 3, 2, 3, 0);
+
+  t8_cmesh_set_join (cmesh, 1, 3, 0, 2, 1);
+  t8_cmesh_set_join (cmesh, 1, 2, 1, 1, 0);
+
+  t8_cmesh_set_join (cmesh, 2, 4, 3, 2, 0);
+  t8_cmesh_set_join (cmesh, 2, 5, 2, 0, 1);
+
+  t8_cmesh_set_join (cmesh, 3, 5, 1, 1, 0);
+  t8_cmesh_set_join (cmesh, 3, 4, 0, 0, 0);
+
+  t8_cmesh_set_join (cmesh, 4, 5, 1, 2, 0);
+
+  /* 7. Commit the mesh */
+  t8_cmesh_commit (cmesh, comm);
+
+  return cmesh;
+}
+
+/* Definition of a three dimensional mesh with linear geometry.
+ * The mesh consists of two tetrahedra, two prisms, one pyramid, and one hexahedron.
+ *  */
+t8_cmesh_t
+t8_cmesh_new_hybrid_gate_3d (sc_MPI_Comm comm)
+{
+  /* Definition of the mesh according to the above scheme
+   * // 1. Defining an array with all vertices
+   * double vertices[ ] =
+   *     { 
+   *       0.43, 0, 2,          // Tetrahedron 1 vertices 
+   0, 0, 1,
+   0.86, -0.5, 1,
+   0.86, 0.5, 1,
+
+   2.29, 0, 2,          // Tetrahedron 2 vertices 
+   1.86, -0.5, 1,
+   2.72, 0, 1, 
+   1.86, 0.5, 1,
+
+   0, 0, 0,             // Prism 1 vertices
+   0.86, -0.5, 0,
+   0.86, 0.5, 0,
+   0, 0, 1, 
+   0.86, -0.5, 1,
+   0.86, 0.5, 1,
+
+   1.86, -0.5, 0,       // Prism 2 vertices
+   2.72, 0, 0,
+   1.86, 0.5, 0,
+   1.86, -0.5, 1,
+   2.72, 0, 1,
+   1.86, 0.5, 1,
+
+   0.86, -0.5, 0,       // Pyramid coordinates
+   1.86, -0.5, 0,
+   0.86, 0.5, 0,
+   1.86, 0.5, 0,
+   1.29, 0, 0,
```

Comment:
```suggestion
   0.86, 0.5, 0,       // Pyramid coordinates
   1.86, 0.5, 0,
   0.86, -0.5, 0,
   1.86, -0.5, 0,
   1.36, 0, -0.5,
```
The vertices of the pyramid within the comment describing the 3D cmesh may be updated as well in order to be consistent, such that the "commented code" produces the same result as the actual code when activated.